### PR TITLE
Week 15 Todo-app - Karolin Albinsson

### PR DESCRIPTION
**Pull request by karolinalbinsson**
Link to my deployed project:

https://jolly-montalcini-e2a7c7.netlify.app/ 

My reflections on how this week's project turned out:
I think it turned out good, I would have liked to spend more time on it.

Things I'd like to have clarified or explained in more detail:
-

### CODE REVIEW

File: code/src/App.js [hunk 1]
```diff
@@ -1,9 +1,30 @@
...
+import { Provider } from 'react-redux';
+import { todos } from './reducers/todos';
+import { combineReducers, createStore } from '@reduxjs/toolkit';
+
+import { Header } from './components/Header';
+import { List } from './components/List';
+
+const reducer = combineReducers({todos : todos.reducer});
+
+const persistedStateJSON = localStorage.getItem("toDoReduxState");
```
Comment (trangtara) [inline code/src/App.js:11]: Snygg!!!

File: code/src/components/EmptyList.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+export const EmptyList = () => {
+  return (
+    <div className="empty-todo-list">
+      <h3 className="empty-todo-list-header">There is nothing here!
+      <span role="img" aria-label="party emoji">🥳</span>
+     </h3>
+     <p className="empty-todo-list-text">Enjoy the peace and quiet or start adding things to your list.</p>
```
Comment (trangtara) [inline code/src/components/EmptyList.js:9]: Hahaha, the message is really nice! I like it.

File: code/src/components/Header.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import moment from "moment";
+
+export const Header =() =>{
+  return (
+    <header className="header">
+      <h1 className="header-title">To-do</h1>
+      <div className="header-sub-wrapper">
+      <h2 className="header-day">{moment().format('dddd')}</h2>
+      <h2 className="header-today-date">{moment().format('LL')}</h2>
```
Comment (trangtara) [inline code/src/components/Header.js:10]: Great codes!

File: code/src/components/List.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+export const List = () =>{
+
+  const itemsList = useSelector((store) => store.todos.list);
+  const listLength = itemsList.items.length;
+  console.log(itemsList.items.length)
+
+  return (
+    <section className="todo-list-wrapper">
+      <TodoInput />
+       {listLength === 0 ? <EmptyList /> : <>
```
Comment (trangtara) [inline code/src/components/List.js:19]: Nice touch with EmptyList

File: code/src/components/TodoInput.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  const[dateValue, setDateValue] = useState(moment().format('YYYY-MM-DD'));
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    dispatch(todos.actions.addToDo({
+      itemInfo: {
+        text: inputValue,
+        done: false,
+        created : moment(),
```
Comment (trangtara) [inline code/src/components/TodoInput.js:19]: It's exciting to learn that you add moment and dueDate in the task. Very smart!

File: code/src/components/TodoInput.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        value ={inputValue} 
+        className = "todo-form-input" 
+        placeholder="Add a new task"></input>
+      </label>
+      <div className="todo-wrapper">
+    
+      <input className="todo-date-picker" type="date" id="dueDate" name="due-date"
+       value={dateValue}
+       min={moment().format('YYYY-MM-DD')} 
+       max={moment().add(1,'years').format('YYYY-MM-DD')}
```
Comment (trangtara) [inline code/src/components/TodoInput.js:42]: Bravo!

File: code/src/components/TodoItem.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  return(
+    <li className={item.done ? "todo-item-done" : "todo-item"}>
+      <div className="item-text-wrapper">   
+      <div className="item-wrapper">
+          <label className="item-label">
+            <input 
+              type="checkbox"
+              onChange={handleOnChange}
+              className="todo-item-checkbox"
+              checked={item.done ? "checked" : "" }
```
Comment (trangtara) [inline code/src/components/TodoItem.js:34]: Nice code with conditional checked attribute

File: code/src/components/TodoItem.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+              >
+            </input>
+          <span className="todo-item-text">{item.text}</span></label>   
+      </div>
+      <div className="item-dates">
+        <span className="todo-date-created">Created: {moment(item.created).fromNow()}</span>
+        <span className={moment(item.dueDate).endOf('day') < moment() ? 'todo-past-due': 'todo-pre-due'}>Due: {moment(item.dueDate).endOf('day').fromNow()}</span>
```
Comment (trangtara) [inline code/src/components/TodoItem.js:41]: Wow! These codes are advanced, Karro. I want to learn them.

File: code/src/index.css [hunk 9]
```diff
@@ -1,13 +1,236 @@
...
+
+.reset-button:hover,
+.todo-item-remove-button:hover {
+	background-color: rgb(149, 5, 94);
+	color: #fff;
+}
+.set-done-button:hover {
+	background-color: rgb(37, 124, 105);
+	color: #fff;
+}
```
Comment (trangtara) [inline code/src/index.css:132]: Cool!

File: code/src/reducers/todos.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const initialState = {
+  list : {
+    name : "TodoList",
+    items : [
+    ]
+  }
```
Comment (trangtara) [inline code/src/reducers/todos.js:8]: It's interesting to see you start with an empty store.

File: code/src/reducers/todos.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    setTaskDone: (state, action) => {
+      const { itemIndex, done }  = action.payload;
+      state.list.items[itemIndex].done = done;
+    },
+    removeTask : (state,action) => {
+      const { itemIndex } = action.payload;
+      state.list.items = state.list.items.filter(
+       (item,index) => 
+        index !== itemIndex
+      )
```
Comment (trangtara) [inline code/src/reducers/todos.js:29]: Wow! I have never used two arguments in side the filter method before. Thank you for showing me how it works.

File: code/src/reducers/todos.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    },
+    removeAllTasks : () => {
+      return initialState;
+    },
```
Comment (trangtara) [inline code/src/reducers/todos.js:33]: Very neat Action. Another pro step you made.

**Review by trangtara — APPROVED** [review]
Hi Karro,
It's a good learning experience for me while reviewing your codes. You have achieved the requirements of the red level(except the styled-components) and some parts of the black level too. Congrats on your good work!
I could not give much comment on your areas for improvement as I see your codes are advanced. I hope Maks can add in some from his experience.
Nice job!
Tara

File: code/src/components/List.js [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+import { TodoItem } from './TodoItem';
+import { TodoSummary } from './TodoSummary';
+import { EmptyList } from './EmptyList';
+import { ListActionBar } from './ListActionBar';
+
+export const List = () =>{
+
+  const itemsList = useSelector((store) => store.todos.list);
+  const listLength = itemsList.items.length;
+  console.log(itemsList.items.length)
```
Comment (HIPPIEKICK) [inline code/src/components/List.js:14]: Remove 😇

File: code/src/components/TodoInput.js [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+        required
+        type="text" 
+        onChange={e => setInputValue(e.target.value)} 
+        value ={inputValue} 
+        className = "todo-form-input" 
+        placeholder="Add a new task"></input>
+      </label>
+      <div className="todo-wrapper">
+    
+      <input className="todo-date-picker" type="date" id="dueDate" name="due-date"
```
Comment (HIPPIEKICK) [inline code/src/components/TodoInput.js:39]: Give all attributes new lines for consistency and readability :)

File: code/src/reducers/todos.js [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+    setAllTasksDone : (state) => {
+        state.list.items = state.list.items.map(item => {
+            return{
+              ...item,
+              done : true
+            }
+        }
+        
+          );
```
Comment (HIPPIEKICK) [inline code/src/reducers/todos.js:42]: Errant semi colon

**Pull request closed without merging**